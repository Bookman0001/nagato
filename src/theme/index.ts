import { createGlobalStyle } from 'styled-components'
import { COLOR } from './constants'

export const GlobalStyle = createGlobalStyle`
  body {
    font-family: system-ui, sans-serif;
    color: ${COLOR.BLACK};
    margin: 0;
    border: 0;
    padding: 0;

    div {
      .tl-edges {
        overflow-x : initial;
      }
    }
    h1 {
      font-family: initial;
      margin: 0;
    }
    h2 {
      font-family: initial;
      margin: 0;
    }
    h3 {
      font-family: initial;
      margin: 0;
    }
    p {
      margin: 0;
    }
    a {
      text-decoration: none;
      color: ${COLOR.BLUE};
      box-shadow : initial;
      cursor: pointer;
    }
    button {
      color: ${COLOR.BLACK};
      cursor: pointer;
    }
    select {
      color: ${COLOR.BLACK};
    }
    input {
      padding: 0;
    }
  }
`

export const DarkMode = createGlobalStyle`
  @media (prefers-color-scheme: dark) {
    body {
      background-color: ${COLOR.BLACK};
      color: ${COLOR.WHITE};
    }
  }
`
