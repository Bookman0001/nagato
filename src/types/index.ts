/* API Response Type Definition */
export type ArticlesResponse = {
  readonly contents: Array<ArticleResponse>
  readonly totalCount: number
  readonly offset: number
  readonly limit: number
}

export type ArticleResponse = {
  readonly id: string
  readonly createdAt: string
  readonly updatedAt: string
  readonly publishedAt: string
  readonly title: string
  readonly description: string
  readonly content: string
}

/* App Type Definition */
export type ArticlesPagination = {
  readonly contents: Array<Article>
  readonly totalCount: number
  readonly offset: number
  readonly limit: number
}

export type Article = {
  readonly id: string
  readonly publishedAt: string
  readonly title: string
  readonly description: string
  readonly content: string
}

export type Params = {
  limit: number
}

export type SearchParams = {
  keyword: string
  page: string
}
