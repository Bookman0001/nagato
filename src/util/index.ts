export const regExp = {
  emailExp:
    /^[a-zA-Z0-9_.+-]+@([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]*\.)+[a-zA-Z]{2,}$/,
}
