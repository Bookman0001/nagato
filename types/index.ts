export type Blogs = {
  posts: {
    title: string
    slug: string
    published: string
    description: string
  }[]
}
