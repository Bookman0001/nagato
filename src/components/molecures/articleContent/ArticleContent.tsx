import Link from 'next/link'
import styled from 'styled-components'

import { color, deviceWidth, fontSize } from 'src/theme/constants'
import type { ContentInArticles } from 'src/types'
import { pagesPath } from 'src/utils/$path'

interface Props {
  article: ContentInArticles
}

export function ArticleContent({ article }: Props) {
  return (
    <Container>
      <Link href={pagesPath.posts._id(article.id).$url()} passHref>
        <StyledLink>
          <SubTitle key={article.id}>{article.title}</SubTitle>
          <DateMemo>{article.publishedAt}</DateMemo>
        </StyledLink>
      </Link>
    </Container>
  )
}

const Container = styled.div`
  a {
    color: ${color.WHITE};
  }
`

const StyledLink = styled.div`
  cursor: pointer;
  line-height: 1.5;
  :hover {
    opacity: 0.7;
  }
`

const SubTitle = styled.h3`
  font-size: ${fontSize.S};
  margin-right: auto;
  text-decoration: underline;
  @media (max-width: ${deviceWidth.PHONE}) {
    font-size: ${fontSize.XS};
  }
`

const DateMemo = styled.div`
  text-align: right;
  line-height: 1.7;
`
