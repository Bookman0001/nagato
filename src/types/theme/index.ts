import { BUTTON_BG_COLOR } from 'src/theme/constants'
import type { valueOf } from 'src/types/util'

export type ButtonBgColor = valueOf<typeof BUTTON_BG_COLOR>
