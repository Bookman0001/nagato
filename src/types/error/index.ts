import { valueOf } from 'src/types/util'

/* Error Definition */
export const errorCode = {
  badRequest: 'ERR400',
  wrongMethodRequest: 'ERR405',
  internalServerError: 'ERR500',
} as const

type ErrorCode = valueOf<typeof errorCode>

type Issues = Array<{ fieldName?: string; message: string }>

export type Error = Readonly<{
  errorCode: ErrorCode
  issues: Issues
}>
