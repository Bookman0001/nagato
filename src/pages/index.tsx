import styled from 'styled-components'
import media from 'styled-media-query'
import { useTranslation } from 'react-i18next'

import { ArticleContents } from 'src/types'
import { articlesController } from 'src/controller/articles'
import { GlobalStyle } from 'src/theme'
import Introduction from 'src/components/organisms/introduction'
import Oss from 'src/components/organisms/oss'
import EnginnerCareer from 'src/components/organisms/enginnerCareer'
import Articles from 'src/components/organisms/articles'
import Header from 'src/components/organisms/header'
import Footer from 'src/components/organisms/footer'
import BackGroundPicture from 'src/components/atoms/backgroundPicture'

interface Props {
  articles: ArticleContents
}

export async function getStaticProps() {
  const { getLimitedArticles } = articlesController()
  const articles = await getLimitedArticles({ limit: 6 })
  return {
    props: {
      articles: articles,
    },
  }
}

export default function Home({ articles }: Props) {
  const { t } = useTranslation()

  return (
    <>
      <GlobalStyle />
      <title>{t('title')}</title>
      <Header />
      <PictureContainer>
        <BackGroundPicture src="/assets/overview.jpg" alt="overview" />
      </PictureContainer>
      <Container>
        <Introduction />
        <Oss />
        <EnginnerCareer />
        <Articles articles={articles} />
      </Container>
      <Footer />
    </>
  )
}

const Container = styled.main`
  max-width: 768px;
  margin: 0 auto;
  padding: 30px 0;
  ${media.lessThan('medium')`
    padding: 30px;
  `}
`

const PictureContainer = styled.div`
  margin: 50px 0 70px;
  ${media.lessThan('medium')`
    margin: 50px 0 20px;
  `}
`
