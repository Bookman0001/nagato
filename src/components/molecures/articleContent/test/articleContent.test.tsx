/**
 * @jest-environment jsdom
 */
import { shallow } from 'enzyme'

import ArticleContent from '../'
import { Article } from 'src/types'

test('ArticleContent', () => {
  const article: Article = {
    id: '1',
    publishedAt: '2021/05/03',
    title: 'タイトル',
    description: 'ここは概要を表示します',
    content: 'サンプルサンプルサンプルサンプル',
  }
  const component = shallow(<ArticleContent article={article} />)
  expect(component).toMatchSnapshot()
})
