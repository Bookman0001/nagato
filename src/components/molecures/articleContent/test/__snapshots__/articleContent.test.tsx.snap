// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContent 1`] = `
<Link
  href="/posts/1"
>
  <styled.div>
    <styled.h3
      key="1"
    >
      タイトル
    </styled.h3>
    <styled.div>
      <ParagraphContent>
        2021/05/03
      </ParagraphContent>
    </styled.div>
  </styled.div>
</Link>
`;
